### PR TITLE
Update tested versions

### PR DESCRIPTION
**Pull request by adamchainz**
I checked the changes made for 4.2 in #131 and extended them.

`tox` passes locally *except* Flake8, which already had a few errors. I can fix those in a follow-up PR.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,42 +1,34 @@
...
+    {py310,py311,py312}-django42
     flake8
 skip_missing_interpreters = True
 
 [flake8]
 max-line-length = 120
 
 [testenv]
-commands = python ./manage.py test
+commands = python -W error::DeprecationWarning -W error::PendingDeprecationWarning ./manage.py test
```
Comment (adamchainz) [inline tox.ini:16]: This pattern helps prevent deprecation warnings from affecting users. I use these flags in my open source projects.

File: tox.ini [hunk 2]
```diff
@@ -1,42 +1,34 @@
...
-    django31: django>=3.1,<3.2
     django32: django>=3.2,<3.3
     django40: django>=4.0,<4.1
     django41: django>=4.1,<4.2
     django42: django>=4.2,<4.3
+    django50: django>=5.0a1,<5.1
 
 [testenv:flake8]
-basepython = python3.7
+basepython = python3.12
```
Comment (adamchainz) [inline tox.ini:30]: Upgrading to the latest whilst at it

File: tox.ini [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
 deps =
-    flake8==3.8.4
+    flake8==6.1.0
```
Comment (adamchainz) [inline tox.ini:32]: Latest Flake8: mostly performance improvements and compatibility fixes. I don't think any of the errors I see are from new rules:

```
./django_lifecycle/dataclass_validation.py:31:26: F541 f-string is missing placeholders
./django_lifecycle/mixins.py:245:5: E303 too many blank lines (2)
./pypi_submit.py:4:33: W292 no newline at end of file
./tests/testapp/tests/test_mixin.py:103:121: E501 line too long (121 > 120 characters)
./tests/testapp/tests/test_mixin.py:104:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:110:121: E501 line too long (122 > 120 characters)
./tests/testapp/tests/test_mixin.py:111:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:128:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:128:121: E501 line too long (146 > 120 characters)
./tests/testapp/tests/test_mixin.py:134:121: E501 line too long (122 > 120 characters)
./tests/testapp/tests/test_mixin.py:135:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:350:25: E251 unexpected spaces around keyword / parameter equals
./tests/testapp/tests/test_mixin.py:350:27: E251 unexpected spaces around keyword / parameter equals
./tests/testapp/tests/test_mixin.py:354:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:355:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:361:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:362:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:368:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:369:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:375:121: E501 line too long (124 > 120 characters)
./tests/testapp/tests/test_mixin.py:376:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:387:121: E501 line too long (150 > 120 characters)
./tests/testapp/tests/test_user_account.py:130:121: E501 line too long (129 > 120 characters)
```

File: tests/testapp/tests/test_mixin.py [hunk 4]
```diff
@@ -339,9 +338,9 @@ def test_comparison_state_should_reset_after_save(self):
         account = UserAccount.objects.create(**data)
         account.first_name = "Maggie"
         self.assertTrue(account.has_changed("first_name"))
-        with capture_on_commit_callbacks(execute=True) as callbacks:
+        with self.captureOnCommitCallbacks(execute=True) as callbacks:
             account.save()
-        self.assertEquals(len(callbacks), 1, msg="Only the _reset_initial_state should be in the on_commit callbacks")
```
Comment (adamchainz) [inline tests/testapp/tests/test_mixin.py:344]: `assertEquals` was deprecated on Python 3.2 and finally removed in 3.12: https://docs.python.org/3.12/whatsnew/3.12.html#id3

File: README.md [hunk 5]
```diff
@@ -61,6 +64,12 @@ Instead of overriding `save` and `__init__` in a clunky way that hurts readabili
 
 # Changelog
 
+## Pending
+
+- Drop support for Python < 3.8.
+- Drop support for Django < 3.2.
```
Comment (EnriqueSoria) [inline README.md:70]: Does it makes sense to remove support for those versions? If we keep `django-capture-on-commit-callbacks==1.10.0` we could still support Django 2.x

**Comment by EnriqueSoria** [comment]
Thanks for this! I've made some changes over your work to keep supporting Python 3.7 and Django 2.2, but thanks to you we have now `tox` in a beautiful green color!

File: README.md (hunk 5 above)
Comment (adamchainz) [reply README.md:70]: I prefer to drop versions that are EOL. Uesrs on old versions can stick to the old versions of the package.

**Pull request merged**